server/src/main/java/org/opensearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -203,8 +203,9 @@ public Object value() {
         return maybeConvertToString(this.value);
     }
 
-    public FuzzyQueryBuilder fuzziness(Fuzziness fuzziness) {
-        this.fuzziness = (fuzziness == null) ? DEFAULT_FUZZINESS : fuzziness;
+    /** Sets the fuzziness used when evaluated to a fuzzy query type. Defaults to "AUTO". */
+    public FuzzyQueryBuilder fuzziness(Object fuzziness) {
```

Comment:
@noCharger I would argue that from the developer's perspective, using `Object` is very discouraging it applies literally "any reference" (in Java). The `Fuzziness.build` as the only way to have `Fuzziness` instances is no better in these regards, the only benefit I see is the opportunity to use `Fuzziness.AUTO` and other constants. 

What do you think if we introduce the builder for `Fuzziness` and make it public API? The idea is instead of manipulating objects have type safe way constructing  `Fuzziness` instances.

Reply:
Looking at the different kinds of `Fuzziness`, they seem to fall into three buckets:
* Fixed edit distance of 0,1, or 2.
* "Custom auto", where you specify a `lowLength` and `highLength`, where the edit distance is 0 for strings shorter than `lowLength`, 1 for strings where `lowLength <= length <= highLength`, and 2 for strings longer than `highLength`.
* "Default auto", which is equivalent to custom auto where lowLength=3 and highLength=6.

There are public constants in `Fuzziness` that cover the fixed edit distance and default auto cases::
```
    public static final Fuzziness ZERO = new Fuzziness(0);
    public static final Fuzziness ONE = new Fuzziness(1);
    public static final Fuzziness TWO = new Fuzziness(2);
    public static final Fuzziness AUTO = new Fuzziness("AUTO");
```

So, the only kind of `Fuzziness` object that we can't just reference as a constant is custom auto. 

What if we just add a static method to `Fuzziness` to cover that use-case:
```
   public static Fuzziness customAuto(int lowLength, in highLength) {
       return new Fuzziness("AUTO", lowLength, highLength);
   }
```

Reply:
Thanks @reta and @msfroh for the great suggestions!

As distance is defined here https://github.com/opensearch-project/OpenSearch/blob/ce25dec68521cc3ee13d42fe52b5912f4448ae71/server/src/main/java/org/opensearch/common/unit/Fuzziness.java#L200-L212, I would agree that there's no other use case except this `customAuto ` above.